### DISCUSSION TITLE
query active variant with extension api

### DISCUSSION THREAD

**Discussion by haai** [microsoft/vscode-cmake-tools#3352] [General]
Hi all.

Is it possible to query the active variant that is set in vscode/cmake-tools extension?
Got a requirement for an extension but this information is needed. Did not find a possibility to query this properly.

Does such an api exist or use case already?

Thanks
haai

**Comment by gcampbell-msft** [comment]
There is a supported official API here: https://www.npmjs.com/package/vscode-cmake-tools?activeTab=code. 

You should be able to hook into an event for when the configuration is changed, and you can also get the active folder and get the project of that path.
